lib/ifconfig/sunos/ifconfig.rb

```diff
@@ -31,7 +31,8 @@ def initialize(input=nil,netstat=nil,verbose=nil)
         when /\s+ether\s+/im
           @ifaces[iface_name] = EthernetAdapter.new(iface_name,iface,netstat)
         else
-          puts "Unknown Adapter Type: #{iface}" if @verbose
+          # fallback for virtual nics like bge:0
+          @ifaces[iface_name] = EthernetAdapter.new(iface_name,iface,netstat)
```

Comment:
The Router flag (indicates that IP packets can be forwarded to and from the interface) belongs to all non loopback interfaces. [Solaris ifconfig man page](http://docs.oracle.com/cd/E19253-01/816-5166/6mbb1kq31/) has it all in. I do not think there is a need for an extra subclass, because we are already inside the Solaris NetworkAdapter class.